Add the ability to make it so tags can be added to all datapoints from t...

kylebrandt opened the pull request:
...he CLI/conf. Fixes #153

A few things I noticed that concern me:
- opentsdb.TagSet.Merge is destructive but also returns the tagset. That threw me off, is this normal?
- Setting the hostname and and now these additional tags is duplicated in both program.go and AddTS, should this be refactored now?

madelynnblue commented on collectors/collectors.go:
```diff
@@ -123,6 +124,8 @@ func AddTS(md *opentsdb.MultiDataPoint, name string, ts int64, value interface{}
 	} else if host == "" {
 		delete(tags, "host")
 	}
+	at := AddTags.Copy()
```
`tags = AddTags.Copy().Merge(tags)`

madelynnblue commented on collectors/program.go:
```diff
@@ -159,6 +159,8 @@ Loop:
 			}
 		}
+		at := AddTags.Copy()
+		dp.Tags = at.Merge(dp.Tags)
```
dpTags = AddTags.Copy().Merge(dp.Tags)

madelynnblue commented:
opentsdb.TagSet.Merge is working exactly as it is documented. So yes, this is normal.

Yes, we should refactor the tag stuff so program.go and AddTS use a similar function. But for now things are probably ok.

kylebrandt committed:
Add the ability to make it so tags can be added to all datapoints from the CLI/conf. Fixes #153

A few things I noticed that concern me:
 - opentsdb.TagSet.Merge is destructive but also returns the tagset. That threw me off, is this normal?
 - Setting the hostname and and now these additional tags is duplicated in both program.go and AddTS, should this be refactored now?
